src/gordon_gcp/plugins/service/enricher.py

```diff
@@ -219,14 +242,16 @@ def _get_internal_ip(self, instance_data):
         msg_logger = _utils.GEventMessageLogger(
             self._logger, {'msg_id': event_message.msg_id})
 
-        instance_data = await self._poll_for_instance_data(
-            event_message.data['resourceName'], msg_logger)
-        records = await self._create_rrecords(
-            event_message.data, instance_data, msg_logger)
-
+        if event_message.data['action'] == 'additions':
+            instance_data = await self._poll_for_instance_data(
+                event_message.data['resourceName'], msg_logger)
+            records = await self._create_rrecords(
+                event_message.data, instance_data, msg_logger)
+        elif event_message.data['action'] == 'deletions':
+            instance_resource_url = event_message.data['resourceName']
+            records = await self._get_matching_records(instance_resource_url)
         msg_logger.debug(f'Enriched with resource record(s): {records}')
         event_message.data['resourceRecords'].extend(records)
-
-        added_records = event_message.data['resourceRecords']
-        msg = f'Enriched msg with {len(added_records)} resource record(s).'
+        msg = (f"Enriched msg with {len(event_message.data['resourceRecords'])}"
+               " resource record(s).")
         event_message.append_to_history(msg, self.phase)
```

Comment:
Should we be doing something here if there are zero records?

Reply:
I've been thinking about this case, and it feels like the best behavior might be to drop/ack the message and appropriately log/record this somewhere?

We might not want to fully treat it as an error as the actual state == the desired state (i.e. we want no A records for the GCE host and the state in Google Cloud DNS is that there are no A records). However, with the logging/recording, we can keep track of how many times this is happening (because this happening too many times indicates something is wrong with the system).

Any examples of this kind of behavior elsewhere in the codebase I can draw inspiration from?

Reply:
I think that sounds ok, but it might add complexity to the system, since right now the assumption is that only failures would cause a message to be dropped from the channel.  We could leave it as a noop for the publisher, and have it just return early if it sees there are no records to publish.

Reply:
Updated the PR with the approach of having the Publisher treat GEventMessages without resource records as a no-op and have the enricher not throw an Exception in this case.

Reply:
Maybe we should add a test for that in the publisher?

Reply:
Does `test_handle_message_no_resource_records` not cover that ?